src/strategies_and_params/replicastrategy.jl

```diff
@@ -49,58 +49,95 @@ NoStats(N=1) = NoStats{N}()
-tuple of operators, the overlaps are computed for all operators.
+Run `num_replicas` replicas and report overlaps between all pairs of replica vectors. If `operator` is
+not `nothing`, the overlap `dot(c1, operator, c2)` is reported as well. If `operator` is a tuple
+of operators, the overlaps are computed for all operators.
 
 Column names in the report are of the form `c{i}_dot_c{j}` for vector-vector overlaps, and
 `c{i}_Op{k}_c{j}` for operator overlaps.
 
 See [`lomc!`](@ref), [`ReplicaStrategy`](@ref) and [`AbstractHamiltonian`](@ref) (for an
 interface for implementing operators).
+
+If `transform` is not `nothing` then this strategy calculates the overlaps with respect to a 
+similarity transformation `G` of the Hamiltonian (see e.g. [`GutzwillerSampling`](@ref).)
+```math
+    \\hat{G} = f \\hat{H} f^{-1}.
+```
+The expectation value of an operator `A` is then
+```math
+    \\langle A \\rangle = \\langle \\psi | A | \\psi \\rangle 
+        = \\frac{\\langle \\phi | f^{-1} A f^{-1} | \\phi \\rangle}{\\langle \\phi | f^{-2} | \\phi \\rangle}
+```
+where 
+```math
+    | \\phi \\rangle = f | \\psi \\rangle
+``` 
+is the (right) eigenvector of ``\\hat{G}``.
+
+For a k-tuple of input operators `(A_1,..., A_k)`, overlaps of ``f^{-1} A_k f^{-1}`` are reported 
+as `c{i}_Op{k}_c{j}`. Overlaps of ``f^{-2}`` are reported *last* as `c{i}_Op{k+1}_c{j}`.
+
+In either case, the untransformed vector-vector overlap `c{i}_dot_c{j}` 
+can be omitted with the flag `vecnorm=false`.
 """
-struct AllOverlaps{N,M,O<:NTuple{M,AbstractHamiltonian}} <: ReplicaStrategy{N}
+struct AllOverlaps{N,M,O<:NTuple{M,AbstractHamiltonian},B} <: ReplicaStrategy{N}
     operators::O
 end
 
-function AllOverlaps(num_replicas=2, operator=nothing)
+function AllOverlaps(num_replicas=2; operator=nothing, transform=nothing, vecnorm=true)
     if isnothing(operator)
         operators = ()
     elseif operator isa Tuple
         operators = operator
     else
         operators = (operator,)
     end
-    return AllOverlaps{num_replicas,length(operators),typeof(operators)}(operators)
+    if isnothing(transform)
+        ops = operators
+    else
+        fsq = Rimu.Hamiltonians.TransformUndoer(transform)
+        ops = (map(op -> Rimu.Hamiltonians.TransformUndoer(transform, op), operators)..., fsq)
+    end    
+    if !vecnorm && length(ops) == 0
+        return NoStats(num_replicas)
+    end
+    return AllOverlaps{num_replicas,length(ops),typeof(ops),vecnorm}(ops)
 end
+@deprecate AllOverlaps(num_replicas, operator) AllOverlaps(num_replicas; operator)
 
-function replica_stats(rs::AllOverlaps, replicas::NTuple{N}) where {N}
+function replica_stats(rs::AllOverlaps{N,<:Any,<:Any,B}, replicas::NTuple{N}) where {N,B}
     # Not using broadcasting because it wasn't inferred properly.
     vecs = ntuple(i -> replicas[i].v, Val(N))
-    return all_overlaps(rs.operators, vecs)
+    return all_overlaps(rs.operators, vecs, B)
```

Comment:
If you are passing `B` here as a regular argument, it becomes runtime information and you loose the benefit of having it as type information in the `AllOverlaps` struct. Moreover, it will make `alloverlaps` and `replica_stats` type unstable. I don't think we want this.

I think the following should work:
```suggestion
    return all_overlaps(rs.operators, vecs, Val{B}())
```